src/app/components/DevSiteRoutingMenu.jsx

```diff
@@ -0,0 +1,211 @@
+     * the desired selection styles are present.
+     */
+    this.state = {
+      selectedChildKey: getSelectedChildKey(props.location.pathname, props.menuItems),
+      items: props.menuItems,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    /**
+     * The selectedChildKey is re-evaluated when new props are received to keep the internal state consistent.
+     */
+    this.setState({
+      selectedChildKey: getSelectedChildKey(nextProps.location.pathname, nextProps.menuItems),
+      items: nextProps.menuItems,
+    });
+  }
+
+  handleMenuChange(_, data) {
+    const { routingStackDelegate, layoutConfig } = this.props;
+
+
+    let routeFunc;
+    if (data.metaData.externalLink) {
+      routeFunc = () => window.open(data.metaData.externalLink.path, data.metaData.externalLink.target || '_blank');
+    } else {
+      this.setState({
+        selectedChildKey: data.selectedChildKey,
+      });
+      routeFunc = () => routingStackDelegate.show({ path: data.metaData.path });
+    }
+
+    /**
+     * If the menuItem does not indicate the presence of a subsequent menu, it is assumed that a terminal
+     * menu item has been selected. If the `toggleMenu` function is defined on the layoutConfig, then it is called
+     * to close the menu before navigating.
+     */
+    if (!data.metaData.hasSubMenu && layoutConfig.toggleMenu) {
+      return layoutConfig.toggleMenu().then(() => {
+        routeFunc();
+      });
+    }
+
+    /**
+     * Otherwise, the layout is left alone and navigation occurs immediately.
+     */
+    return Promise.resolve().then(() => routeFunc());
+  }
+
+  handleFilter(event) {
+    const filterText = event.target.value;
+    const { menuItems } = this.props;
+
+    const filter = searchFilter(filterText);
+    const items = menuItems.filter(({ text }) => filter(text));
+
+    this.setState({ items });
+  }
+
+  render() {
+    const { title, routingStackDelegate } = this.props;
+    const { selectedChildKey, items } = this.state;
+
+    /**
+     * The items using the simplified DevSiteRoutingMenu menuItem API must be converted into the NavigationSideMenu's API.
+     */
+    const processedMenuItems = buildSideMenuItems(items);
+
+    /**
+     * The DevSiteRoutingMenu then constructs a menuItem that will act as the main page and render the menuItems as child items.
+     * If there is no title and no `showParent` implementation present, the isRootMenu flag is set on the item to hide
+     * the header altogether.
+     */
+    processedMenuItems.push({
+      key: devSiteRoutingMenuRootMenuKey,
+      text: title || '',
+      childKeys: processedMenuItems.map(item => item.key),
+      isRootMenu: !routingStackDelegate.showParent && !title,
+    });
+
+    const hasToolbar = (items || []).length >= FILTER_DISPLAY_THRESHOLD;
+    const toolbar = (hasToolbar
+      ? (
+        <FormattedMessage id="Terra.devSiteRoutingMenu.filter">
+          {placeholder => (
+            <SearchField
```

Comment:
Do we want to toggle rendering when a SearchField is available or do we want to disable the SearchField when the number of menu items are less than the threshold?

Reply:
I'm not sure I understand the question..

UX wants the filter to only appear when there are at least 10 items in the nav

Reply:
As you navigate through nested links, does this change when the search field is displayed?

Reply:
For example,
Terra-Core ~= 80 list items, show search field 
> click terra-list
List ~= 3, don't show search field

Reply:
Correct, we don't want to show the search field in the terra-list..

I initially had a typo, but the implementation is to only display the filter iff the initial menu items (`{ menuItems } = this.props`) length >= 10